services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -656,29 +672,35 @@ class AudioColumn(MediaColumn):
     transform_column = FloatColumn
 
     @staticmethod
-    def get_duration(example: dict[str, Any]) -> float:
+    def get_duration(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[float]:
         """Get audio durations"""
-        with io.BytesIO(example["bytes"]) as f:
+        if example is None:
+            return None
+        example_bytes = example["bytes"] if is_struct else example  # type: ignore
+        with io.BytesIO(example_bytes) as f:
```

Comment:
I guess you can just use isinstance ?
```suggestion
        if example is None:
            return None
        elif isinstance(example, bytes):
            example_bytes = example
        else:
            example_bytes = example["bytes"]
        with io.BytesIO(example_bytes) as f:
```

Reply:
yeah I first did like this but then I thought if I know in advance for all samples, why check every time. but if you think it's fine, i'd do like this